### PR TITLE
Menus API and UI

### PR DESCRIPTION
**Pull request by kaushikravikumar**
### Summary
This pull request sets the foundation for implementing menus in the app. The model class for Menus has been created to represent a specific facility's menu on a certain day, containing every meal from breakfast to dinner. This pull request adds a function in the API class, **fetchMenuJSON()**, which makes the get request with appropriate parameters. The FacilityInfo Activity then includes the function which passes in the response handler as a parameter. The response handler then updates the UI accordingly.

This pull request is the first step towards implementing the menus feature for Flux. 

- [x] implemented Model Class for Menus
- [x] set up API request code and response handler function
- [x] created recyclerview to show categories and food items
- [x] created chips to view menu for specific meal of day
- [x] Make Chips toggle, so you can't unselect a chip
- [ ] refine Menus User Interface

### Test Plan 

84665836_209826860170233_8413460480365953024_n

- Test edge cases by viewing menus with many meals of day to see if chips overflow screen.
- Test switching meal of day to see if there is significant lag in loading menus.

### CODE REVIEW

File: Density/app/src/main/java/org/cornelldti/density/density/BaseActivity.kt [hunk 1]
```diff
@@ -61,6 +61,7 @@ open class BaseActivity :
                 .addOnCompleteListener { task ->
                     if (task.isSuccessful) {
                         Log.d("checkpoint", "gotToken")
+                        Log.d("TOKEN1", task.result?.token)
```
Comment (SamChou19815) [inline Density/app/src/main/java/org/cornelldti/density/density/BaseActivity.kt]: Do not log token

File: Density/app/src/main/java/org/cornelldti/density/density/data/MenuClass.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package org.cornelldti.density.density.data
+
+class MenuClass(facility: String) {
+    var facilityId: String? = null
+
+    var breakfastItems: ArrayList<CategoryItems>? = null
+    var brunchItems: ArrayList<CategoryItems>? = null
+    var lunchItems: ArrayList<CategoryItems>? = null
+    var liteLunchItems: ArrayList<CategoryItems>? = null
+    var dinnerItems: ArrayList<CategoryItems>? = null
```
Comment (SamChou19815) [inline Density/app/src/main/java/org/cornelldti/density/density/data/MenuClass.kt]: Avoid using `ArrayList`. Use `List` whenever you can. `List` interface in kotlin is read-only.

File: Density/app/src/main/java/org/cornelldti/density/density/data/MenuClass.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    init {
+        facilityId = facility
+    }
+
+    class CategoryItems(categoryName: String, itemNames: ArrayList<String>) {
```
Comment (SamChou19815) [inline Density/app/src/main/java/org/cornelldti/density/density/data/MenuClass.kt]: Same here, try avoid `ArrayList`

File: Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt [hunk 4]
```diff
@@ -255,6 +260,55 @@ class FacilityInfoPage : BaseActivity() {
...
+
+    private fun showMenu(menu : MenuClass?) {
+        var menuText = ""
+        if(menu != null) {
+            if(menu.breakfastItems != null) {
+                menuText.plus("Breakfast:" + "\n")
+                for(i in 0 until menu.breakfastItems!!.size) {
+                    menuText.plus(menu.breakfastItems!![i].stringify())
+                }
+            }
```
Comment (SamChou19815) [inline Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt]: String concat in loop is efficient. Use `StringBuilder` instead.

File: Density/app/src/main/java/org/cornelldti/density/density/network/API.kt [hunk 5]
```diff
@@ -126,6 +127,7 @@ class API(context: Context) {
             fetchOperatingHoursOnResponse: (operatingHours: List<String>) -> Unit,
             fetchHistoricalJSONOnResponse: (densities: List<Double>) -> Unit
     ) {
+//        Log.d("TOKEN3", idToken)
```
Comment (SamChou19815) [inline Density/app/src/main/java/org/cornelldti/density/density/network/API.kt]: No commented code

File: Density/app/src/main/java/org/cornelldti/density/density/network/API.kt [hunk 6]
```diff
@@ -161,6 +163,22 @@ class API(context: Context) {
         queue.add(facilityRequest)
     }
 
+    fun fetchMenuJSON(
+            facilityId: String,
+            day: String,
+            fetchMenuJSONOnResponse: ((menu: MenuClass?) -> Unit)
+    ) {
+//        Log.d("TOKEN2", idToken)
```
Comment (SamChou19815) [inline Density/app/src/main/java/org/cornelldti/density/density/network/API.kt]: No commented code

File: Density/app/src/main/java/org/cornelldti/density/density/data/MenuClass.kt (hunk 2 above)
Comment (SamChou19815) [reply Density/app/src/main/java/org/cornelldti/density/density/data/MenuClass.kt]: Also, why is everything mutable and public here?

**Comment by kungpaogao** [comment]
Reviewing this right now. Notes so far:
- some code in `init { ... }` methods can be refactored to just the class constructor
- menu chips can be unselected (probably just need to change to a different chip style)
- I think the chips are visible briefly even on places without menus - also we probably shouldn't even show the menu card if there are no menus available

**Comment by kungpaogao** [comment]
Merged `master` and made a lot of interface changes. Now, I just need to make some code design changes, and we should be set to merge :)

**Comment by kungpaogao** [comment]
Merged, but created #42
